src/Web/Rsp.IrasPortal.Web/Controllers/QuestionSetController.cs

```diff
@@ -1,287 +1,215 @@
 ﻿using System.Data;
 using System.Diagnostics.CodeAnalysis;
-using System.Text;
-using ExcelDataReader;
+using System.Text.Json;
 using FluentValidation;
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
 using Rsp.IrasPortal.Application.Constants;
 using Rsp.IrasPortal.Application.DTOs;
 using Rsp.IrasPortal.Application.Services;
+using Rsp.IrasPortal.Web.Extensions;
 using Rsp.IrasPortal.Web.Models;
+using static Rsp.IrasPortal.Application.Constants.QuestionCategories;
 
 namespace Rsp.IrasPortal.Web.Controllers;
 
 [ExcludeFromCodeCoverage]
 [Route("[controller]/[action]", Name = "questionset:[action]")]
 [Authorize(Policy = "IsAdmin")]
-public class QuestionSetController(IQuestionSetService questionSetService, IValidator<QuestionSetFileModel> validator) : Controller
+public class QuestionSetController(IQuestionSetService questionSetService, IValidator<QuestionSetViewModel> validator) : Controller
 {
-    public IActionResult Upload()
+    public async Task<IActionResult> Index(QuestionSetViewModel model)
     {
-        return View();
+        await GetVersions(model);
+
+        return View(model);
+    }
+
+    private async Task GetVersions(QuestionSetViewModel model)
+    {
+        var response = await questionSetService.GetVersions();
+        if (response.IsSuccessStatusCode)
+        {
+            model.Versions = response.Content?.OrderByDescending(x => x.CreatedAt).ToList() ?? [];
+        }
     }
 
     [HttpPost]
-    public async Task<IActionResult> Upload(QuestionSetFileModel model)
+    public async Task<IActionResult> Upload(QuestionSetViewModel model)
     {
         var file = model.Upload;
+        await GetVersions(model);
 
         if (file == null || file.Length == 0)
         {
             ModelState.AddModelError("Upload", "Please upload a file");
```

Comment:
use nameof(Upload) instead of "Upload"
```suggestion
            ModelState.AddModelError(nameof(Upload), "Please upload a file");
```